sdk/tests/wallet/transactions.rs

```diff
@@ -255,3 +255,55 @@ async fn conflicting_transaction() -> Result<()> {
     tear_down(storage_path_0).ok();
     tear_down(storage_path_1)
 }
+
+#[tokio::test]
+#[cfg(all(feature = "ledger_nano", feature = "events"))]
+#[ignore = "requires ledger nano instance"]
+async fn prepare_transaction_ledger() -> Result<()> {
+    use std::sync::{Arc, Mutex};
+
+    use iota_sdk::wallet::events::{types::TransactionProgressEvent, WalletEvent, WalletEventType};
+
+    let storage_path = "test-storage/wallet_address_generation_ledger";
+    setup(storage_path)?;
+
+    let wallet = crate::wallet::common::make_ledger_nano_wallet(storage_path, None).await?;
+
+    let account_0 = &create_accounts_with_funds(&wallet, 1).await?[0];
+    let account_1 = wallet.create_account().finish().await?;
+
+    let amount = 1_000_000;
+
+    let prepared_event = Arc::new(Mutex::new(None));
+    let prepared_event_clone = prepared_event.clone();
+
+    wallet
+        .listen([WalletEventType::TransactionProgress], move |event| {
+            if let WalletEvent::TransactionProgress(progress) = &event.event {
+                *prepared_event_clone.lock().unwrap() = Some(progress.clone());
```

Comment:
I think here it also need to match on the progress `TransactionProgressEvent`, because there are multiple events and the latest will be `TransactionProgressEvent::Broadcasting` wich would then overwrite the `PreparedTransaction` one before